gnome-initial-setup/pages/network/gis-network-page.c

```diff
@@ -588,11 +595,48 @@ active_connections_changed (NMClient *client, GParamSpec *pspec, GisNetworkPage
   refresh_wireless_list (page);
 }
 
+static gboolean
+skip_page_for_ethernet_connection (GisNetworkPage  *self,
+                                   GNetworkMonitor *monitor)
+{
+  GisNetworkPagePrivate *priv;
+  GisAssistant *assistant;
+  gboolean is_current_page;
+
+  priv = gis_network_page_get_instance_private (self);
+  assistant = gis_driver_get_assistant (GIS_PAGE (self)->driver);
+  is_current_page = gis_assistant_get_current_page (assistant) == GIS_PAGE (self);
+
+  if (g_network_monitor_get_connectivity (monitor) != G_NETWORK_CONNECTIVITY_FULL)
+    {
+      if (!is_current_page || priv->row_clicked)
```

Comment:
This function does not always return a value. In the case where connectivity != `G_NETWORK_CONNECTIVITY_FULL`, but `is_current_page && !priv->row_clicked`, then what value should be returned?

(GCC should be issuing a warning for this; I don't think it should be part of this branch but we should make upstream g-i-s compile without warnings and then enable -Werror during development.)